src/service/api.tsx

```diff
@@ -124,6 +128,50 @@ const api = {
     return obj.data.results;
   },
 
+  async getSearchFileredList(dataFilter:
+    {
+      from: number,
+      to: number,
+      gener: string | null,
+      page: number
+    } = {
+      from: 1900,
+      to: 2020,
+      gener: null,
+      page: 1,
+    }
+  ) {
+
+    function setFilteredData() {
+      const { from, to } = dataFilter
+      if (from < to) {
+        return `&primary_release_date.gte=${from}-01-01&primary_release_date.lte=${to}-01-01`
+      } else if (from > to) {
+        return `&primary_release_date.gte=${to}-01-01&primary_release_date.lte=${from}-01-01`
+      }
+      return `&primary_release_date.gte=${from}-01-01`
+    }
+
+    function setGener() {
+      const { gener } = dataFilter;
+      if (gener !== null) {
+        const generId = API_GENER_ID.find(item => item.name === gener);
+        return `&with_genres=${generId?.id}`
+      }
+      return '';
+    }
+
+    let obj = await axios.get(`discover/movie?sort_by=popularity.asc&page=${dataFilter.page}${setFilteredData()}${setGener()}&${API_KEY}`);
+    return obj.data.results;
+  },
+
+  changeImgLindks(arr: MovieCard[], idsList: number[]) {
+    const conformityIds = arr.filter(item => {
+      return idsList.find(itemId => itemId === item.id)
```

Comment:
probably return idsList.includes(item.id) is better here

Reply:
this is what we need))))